etl_manager/meta.py

```diff
@@ -222,8 +239,10 @@ def remove_column(self, column_name):
         self._check_column_exists(column_name)
         new_cols = [c for c in self.columns if c["name"] != column_name]
         new_partitions = [p for p in self.partitions if p != column_name]
+        new_primary_key = [pk for pk in self.primary_key if pk != column_name]
         self.columns = new_cols
         self.partitions = new_partitions
+        self.primary_key = new_primary_key
         self._update_sensitivity()
 
     def add_column(
```

Comment:
@MrAlecJohnson I was thinking about this earlier. This could be done in a similar way to `sensitivity`, whereby the attribute is set at a column level, but there is a table-level attribute that is generated automatically but can't be set directly (to avoid conflicts). Do you think that would be a better way of structuring the information. It would require some rework to the metadata extraction stuff though.

Reply:
Ah, that does look like a neat approach, and makes sense to make it like sensitivity. But despite my comment yesterday, on reflection it feels like premature optimisation. 

So far we haven't seen this being a problem. I also think it's probably low risk - it seems like anywhere using etl_manager to replicate an existing database including its primary keys will also be using the 'read_database_folder' type functions rather than adding columns individually.

Maybe we could note this somewhere as a possible low priority future change?

Reply:
OK, let's leave as is for now. I've created a ticket in the backlog to investigate this in future if we want (PDE-291).